src/dxil/utils.rs

```diff
@@ -0,0 +1,29 @@
+use crate::utils::HassleError;
+
+/// Helper function to validate a DXIL binary independant from the compilation process,
+/// this function expects `dxcompiler.dll` and `dxil.dll` to be available in the current
+/// execution environment.
+/// `dxil.dll` is currently not available on Linux.
+pub fn validate_dxil(data: &[u8]) -> Result<Vec<u8>, HassleError> {
+    let dxc = crate::Dxc::new()?;
+
+    let validator = dxc.create_validator()?;
```

Comment:
Dit was eigenlijk meer een [testje](https://github.com/Traverse-Research/hassle-rs/commit/3935ecae511103a6fd725579cd3505513d2f8fe3) van mij om te zien wat er zou gebeuren. Op Windows word deze call geforward naar dxil en krijg je dezelfde interface; als deze library niet gevonden word, of als je op Linux zit krijg je een shim die de output op 0 laat/zet. Zonder error. Denk dus dat we dat niet willen gebruiken.

Reply:
@DBouma @Jasper-Bekkers Willen we deze change nog terugdraaien?

Reply:
Ik heb deze nu verwijderd gezien het puur een test was, kan hem ook terug zetten op de originele plek?

Reply:
@DBouma Ja hij moet wel weer terug, maar dan met `Dxil::create_validator()` ipv de silently-failing passthrough uit dxc.

Ook gelijk even de relevante tests fixen om `validate_dxil` alleen op Windows aan te roepen.